### PR TITLE
Use Redis for caching

### PR DESCRIPTION
**Pull request by doaa-altarawy**
## Description

Use Redis for caching in Flask, and also in the docker containers. The use of Redis is optional, and by default memory caching is used in the development, but it can be set in config or in environment variable (CACHE_TYPE). Production will use Redis by default.
Max memory is set to 1GB.

## Status
- [x] Ready to go

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,6 +98,32 @@ For a bug with Flask, Keras, and Tensorflow, don't run with reload
 flask run --no-reload --without-threads
 ```
 
+### To try Redis caching locallyin dev, run Redis before running the Flask app:
```
Comment (mattwelborn) [inline README.md]: typo

File: devtools/run-redis.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+if [ ! -d redis-stable/src ]; then
+    curl -O http://download.redis.io/redis-stable.tar.gz
+    tar xvzf redis-stable.tar.gz
+    rm redis-stable.tar.gz
+fi
+cd redis-stable
+make
+src/redis-server --maxmemory 1GB
```
Comment (mattwelborn) [inline devtools/run-redis.sh]: Ask @dgasmith about this one, but I expect that we might need more memory for some of the really big datasets like SSI. (And potentially ACCDB in the future?)

File: docker-compose.yml [hunk 3]
```diff
@@ -13,16 +13,29 @@ services:
...
+      - redis_service
     networks:
       - frontend
       - backend
 
+  redis_service:
+    container_name: redis
+    image: redis
+    restart: unless-stopped
+    command: redis-server --maxmemory 1GB
```
Comment (mattwelborn) [inline docker-compose.yml]: Same comment as above, though it makes more sense here...

File: docker-compose.yml [hunk 4]
```diff
@@ -13,16 +13,29 @@ services:
     restart: unless-stopped
     environment:
       # set by a .env file (add to production)
+      - FLASK_CONFIG=production
       - MONGO_URI=$MONGO_URI
 #      - MONGO_URI=mongodb://doaa:doaapass@mongo_service:27017/qca_logging_db
+      - CACHE_REDIS_URL=redis://redis_service:6379/0
```
Comment (dgasmith) [inline docker-compose.yml]: `CACHE_TYPE=redis`?

Comment (doaa-altarawy) [reply docker-compose.yml]: No need, production, including docker, has Redis cache by default.

File: config.py [hunk 5]
```diff
@@ -67,9 +66,10 @@ class TestingConfig(Config):
 
 
 class ProductionConfig(Config):
-    # SQLALCHEMY_DATABASE_URI = os.environ.get('DATABASE_URL') or \
-    #     'sqlite:///' + os.path.join(basedir, 'data.sqlite')
-    pass
+
+    # Default flask-caching to redis in prod
+    CACHE_TYPE = os.environ.get('CACHE_TYPE') or 'redis'
```
Comment (dgasmith) [inline config.py]: Ah, I see.

File: app/dash_apps/reaction_viewer/index.py [hunk 6]
```diff
@@ -22,7 +22,6 @@
 logger = logging.getLogger(__name__)
 
 SHOW_TIME = False
-CACHE_TIMEOUT = 3600 * 24 * 60  # Two months, effectively no timeout
```
[the remark this answers is not in the document]
Comment (doaa-altarawy) [reply app/dash_apps/reaction_viewer/index.py]: ok

**Pull request merged**